### PR TITLE
Install gcc-c and gcc-c++ for solaris11

### PR DESCRIPTION
**Pull request by jaymalasinha**
Signed-off-by: Jaymala Sinha <jsinha@chef.io>

### Description

The gcc-c and gcc-c++ packages need to installed in addition to gcc for gcc-4 on solaris 11.
Removing gcc-3 as we use gcc-4 now 

### Issues Resolved

[List any existing issues this PR resolves]

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

**Comment by jaymalasinha** [comment]
@scotthain @jeremiahsnapp

File: resources/build_essential.rb [hunk 1]
```diff
@@ -62,11 +62,13 @@
       package 'bison'
       package 'gnu-coreutils'
       package 'flex'
-      package 'gcc' do
-        # lock because we don't use 5 yet
-        version '4.8.2'
+      # lock gcc versions because we don't use 5 yet
+      %w(gcc gcc-c gcc-c++).each do |pkg|
```
Comment (jeremiahsnapp) [inline resources/build_essential.rb]: I think one of the Travis failures is caused by foodcritic rule getting triggered here. I think you might have to just define each package as a separate resource.

File: spec/recipes/solaris2_spec.rb [hunk 2]
```diff
@@ -13,7 +13,7 @@
     expect(chef_run).to install_package('gnu-coreutils')
     expect(chef_run).to install_package('flex')
     expect(chef_run).to install_package('gcc').with_version('4.8.2')
-    expect(chef_run).to install_package('gcc-3')
+    expect(chef_run).to install_package('gcc-c').with_version('4.8.2')
```
Comment (jeremiahsnapp) [inline spec/recipes/solaris2_spec.rb]: should there also be a unit test for `gcc-c++`?

File: resources/build_essential.rb (hunk 1 above)
Comment (tas50) [reply resources/build_essential.rb]: You can inline disable this foodcritic rule as well

File: spec/recipes/solaris2_spec.rb (hunk 2 above)
Comment (jaymalasinha) [reply spec/recipes/solaris2_spec.rb]: Added

**Comment by jaymalasinha** [comment]
@jeremiahsnapp @tas50 Addressed review comments and travis happy now.  Please review the update.  Thanks

File: resources/build_essential.rb [hunk 3]
```diff
@@ -27,6 +27,9 @@
   when 'debian'
     package %w( autoconf binutils-doc bison build-essential flex gettext ncurses-dev )
   when 'amazon', 'fedora', 'rhel'
+    file '/etc/resolv.conf' do
+      content 'nameserver 8.8.8.8'
+    end if node['platform_family'] == 'amazon'
```
Comment (jeremiahsnapp) [inline resources/build_essential.rb]: Would an `only_if` be more appropriate here?

Comment (scotthain) [reply resources/build_essential.rb]: I'm not sure if this is the best way to fix it - I think this was fixed here: https://github.com/someara/dokken-images/pull/14 we just need to consume it?

**Comment by jaymalasinha** [comment]
Thank You @jeremiahsnapp for the dns fix  👍

**Pull request merged**